content/en/docs/Advanced/revoking-secureboot-access.md

```diff
@@ -0,0 +1,300 @@
+  --keys /keys \
+  --output-type container \
+  --boot-branding "KairosNewUKI"
+```
+
+#### Enroll the new certificate in db
+
+From withing Kairos (still booted in the old image), enroll the new db key:
+
+```bash
+sudo chattr -i /sys/firmware/efi/efivars/{PK,KEK,db}*
+efi-updatevar -f db.auth db
+```
+
+{{% alert title="Warning" %}}
+You will need to `efitools` installed on the Kairos image for this commands to work!
+{{% /alert %}}
+
+#### Enroll the old certificate in dbx
+
+As describe earlier in this document, you should generate a `db-dbx.auth` file
+from the old db key. This should now be enrolled in dbx. In user space again
+(booted in the old Kairos image):
+
+```bash
+sudo efi-updatevar -f db-dbx.auth dbx
+```
+
+
+#### Upgrade to the new image
+
+```
+kairos-agent upgrade --source oci:myimage
+```
+
+#### Unbind PCR 7
+
+Bind partition encryption only to PCR 11 policy. From within Kairos again:
+
+```bash
+echo "Generating temporary passphrase"
+dd if=/dev/random bs=32 count=1 of=/tmp/random_keyfile
+
+echo "Adding password slot"
+cryptsetup luksAddKey --token-type systemd-tpm2 /dev/vda2 /tmp/random_keyfile
+cryptsetup luksAddKey --token-type systemd-tpm2 /dev/vda3 /tmp/random_keyfile
+
+echo "Removing the tpm2 slot"
+systemd-cryptenroll /dev/vda2 --wipe-slot=tpm2
+systemd-cryptenroll /dev/vda3 --wipe-slot=tpm2
+
+echo "Adding tpm2 slot again (pcr 11 policy only, no pcr 7)"
+systemd-cryptenroll --unlock-key-file=/tmp/random_keyfile --tpm2-public-key=/run/systemd/tpm2-pcr-public-key.pem --tpm2-public-key-pcrs=11 --tpm2-pcrs= --tpm2-signature=/run/systemd/tpm2-pcr-signature.json --tpm2-device-key=/run/systemd/tpm2-srk-public-key.tpm2b_public /dev/vda2
+systemd-cryptenroll --unlock-key-file=/tmp/random_keyfile --tpm2-public-key=/run/systemd/tpm2-pcr-public-key.pem --tpm2-public-key-pcrs=11 --tpm2-pcrs= --tpm2-signature=/run/systemd/tpm2-pcr-signature.json --tpm2-device-key=/run/systemd/tpm2-srk-public-key.tpm2b_public /dev/vda3
+
+echo "Removing the password slot"
+systemd-cryptenroll --wipe-slot=password /dev/vda2
+systemd-cryptenroll --wipe-slot=password /dev/vda3
+```
+
+Make sure you use the correct device paths for your encrypted partitions
+(`/dev/vda2` and `/dev/vda3` in the example).
+
+Now reboot:
+
+```bash
+reboot
+```
+
+#### Bind again to PCR 11 and 7
+
+After rebooting to the upgraded image, bind the encryption again to PCR 7 (and 11):
+
+
+```bash
+echo "Generating temporary passphrase"
+dd if=/dev/random bs=32 count=1 of=/tmp/random_keyfile
+
+echo "Adding password slot"
+cryptsetup luksAddKey --token-type systemd-tpm2 /dev/vda2 /tmp/random_keyfile
+cryptsetup luksAddKey --token-type systemd-tpm2 /dev/vda3 /tmp/random_keyfile
+
+echo "Removing the tpm2 slot"
+systemd-cryptenroll /dev/vda2 --wipe-slot=tpm2
+systemd-cryptenroll /dev/vda3 --wipe-slot=tpm2
+
+echo "Adding tpm2 slot again (pcr 11 policy AND pcr 7)"
+echo "There is probably a bug in systemd preventing us from using the --tpm2-device-key"
```

Comment:
Weird. It should have the public key in there always?

Reply:
where? Are you referring to the commands below? I pass the public key, yes. Do you mean I shouldn't?

Reply:
in the machine I mean, you mention a bug preventing from using the device key, but its not clear to me what the issue is or why using the tpm-device instead makes it work?

skipping hitting the tpm device should be the way to go always, so we prevent blocking the tpm AND we can choose the PCR numbers that we want. Using --device-auto should be the last resort only done on systemd <255

Reply:
Well, it was you that suggested that this is a bug in systemd, remember? :D And actually you suggested the command that works.

Reply:
yes, but thats for enrolling when we DONT want to bind to PCR7, when we want to do so, we can use the 

Basically:

 - using` --tpm2-device` makes it so the PCR7 is always binded no matter what flags you use
 - using `--tpm2-device-key` makes it so we can bind to anything, even leave PCR bind empty
 - using `--tpm2-device-key` also does not hit the TPM at all, which is nice as we can end up blocking it
 - `--tpm2-device-key` can only be used on systemd 255 (buntu 24.04 and Fedora 40)

So we should favor using always the `--tpm2-device-key` flag with its other `--tpm2-pcrs` and `--tpm2-public-key-pcrs` flags to have full control over whats binding or not.

And because we need to use that `--tpm2-device-key` in order to unbind from PCR7 we should only mention that flag and ignore the `--tpm2-device` one

Reply:
This is the error I'm getting:

```
root@localhost:/home/kairos# systemd-cryptenroll --unlock-key-file=/tmp/random_keyfile --tpm2-public-key=/run/systemd/tpm2-pcr-public-key.pem --tpm2-public-key-pcrs=11 --tpm2-pcrs=7 --tpm2-signature=/run/systemd/tpm2-pcr-signature.json --tpm2-device-key=/run/systemd/tpm2-srk-public-key.tpm2b_public /dev/vda2
/tmp/random_keyfile has 0644 mode that is too permissive, please adjust the ownership and access mode.
Must provide all PCR values when using TPM2 device key.
```

Reply:
ok, we decided to keep it as is for now.